### ISSUE TITLE
argparse.Namespace is more complicated than it seems

### ISSUE THREAD

**Issue by gvanrossum** [python/typeshed#25]
The `argparse.Namespace` class stub doesn't reflect its actual complexity -- it has dynamically created attributes based on the parser specification. For example:
```
import argparse
parser = argparse.ArgumentParser()
parser.add_argument('-d', '--database', type=str, default='data.db')
def foo(args: argparse.Namespace):
    print(args.database)
def main():
    args = parser.parse_args()
    assert args.__class__ is argparse.Namespace
    foo(args)
```
This gives these errors:
```
t.py: note: In function "foo":
t.py:5: error: "Namespace" has no attribute "database"
```
Now, ideally there would be some kind of plugin that understands the parser specification. But I'd be happy if it just didn't complain about using unknown attributes of `Namespace` objects, while still verifying that the caller indeed passes something of that type (so I don't want to just use `Any`).

**Comment by matthiaskramm** [comment]
It sounds like you'd want something like
```
class Namespace:
  def __getattr__(self, s: str) -> Any
```
?

**Comment by gvanrossum** [comment]
Would that work?

**Comment by matthiaskramm** [comment]
It works in mypy (I just tried it). It will also work in pytype as soon as I get around to implementing __getattr__ support.

**Comment by o11c** [comment]
There are already two types that must necessarily support per-instance attributes with full typechecking: classes and modules.

**Comment by gvanrossum** [comment]
This was closed by 9ae0c0b3375b95dedd15a83786a7228b19f00873.

**Comment by mo22** [comment]
For anyone reaching this from google with the same problem as me (namespace is Any), I've build a small script around this where you define arguments as class:
```
    class ArgsImportYml(args2.Command):
        _name = 'import-yml'
        group = Argument(Optional[List[str]], help='name of group(s)')
        zone = Argument(Optional[List[str]], help='name of zone(s)')
        test = Argument(Optional[str])
        dry_run = Argument(bool)
        create = Argument(bool, help='create route53 zone if needed')
        files = Positional(List[args2.ReadableTextFile], help='path to yml file(s)')
```

https://gist.github.com/mo22/3c5cc239b6237d2a3ee2b86eb6960325
